### PR TITLE
Update date indexing and display

### LINKED ISSUE #551: Support date qualifiers in display
Story:
Dates are sometimes given as approximations using the “qualifier” attribute. Exhibit curators would like to provide an accurate representation of date uncertainty for users.

Acceptance criteria:
- Single date display should include qualifier information. (This reproduces current SearchWorks behavior.)
- Date range display should include qualifier information. (This reproduces current SearchWorks behavior.)

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #551 
Partially address #548 

- Uses the new stanford-mods imprint logic
- Add a new date field for display that includes qualifiers and ranges
- Fixes incorrect indexing of `1uuu` dates

### CODE REVIEW

File: lib/traject/dor_config.rb [hunk 1]
```diff
@@ -79,6 +79,19 @@
...
 to_field 'pub_year_w_approx_isi', stanford_mods(:pub_year_int, false)
+
+to_field 'date_display' do |resource, accumulator, _context|
+  imprint = ModsDisplay::Imprint.new(nil, ModsDisplay::Configuration::Imprint.new, nil)
+
+  Array(resource.smods_rec.origin_info).map do |value|
+    dates = imprint.dates(value)
+    accumulator.concat(dates.map(&:values).flatten)
+
+    part = imprint.send(:parts_element, value)
```
Comment (drh-stanford) [inline lib/traject/dor_config.rb]: is this calling a private method in the Imprint class? should we rework mods_display to make that a public method?

Comment (cbeer) [reply lib/traject/dor_config.rb]: Yes, we should. It's going to be an unfortunate amount of work, though, to refactor that code and reconcile it with stanford-mods (which seems to have a copy/pasted version of the code, which was later tweaked).

**Comment by camillevilla** [comment]
@cbeer I'm having difficulty getting this to work locally. I tried adding the two druids provided in #551 and #548 (tw368kj5707, rx795tx1655). Item status shows they both throw the following error: 
```
#<RSolr::Error::Http: RSolr::Error::Http - 400 Bad Request Error: {"responseHeader":{"status":400,"QTime":1},"error":{"metadata":["error-class","org.apache.solr.common.SolrException","root-error-class","org.apache.solr.common.SolrException"],"msg":"ERROR: [doc=rx795tx1655] unknown field 'date_display'","code":400}} 
...
```

Not sure if this is a problem with my dev environment, but I would appreciate someone else giving this a test drive.

**Comment by mejackreed** [comment]
@camillevilla I've added a commit that adds the needed field in the solr schema.xml. I've also created a PR to sul-solr-configs https://github.com/sul-dlss/sul-solr-configs/pull/107 that when merged by devops will update our prod/stage config.

**Comment by cbeer** [comment]
Ugh. Sorry about that. I wonder if we want to switch from `_display` to one of the `*_ssim` fields?

**Comment by mejackreed** [comment]
@cbeer Yeah I'm good with that. I'll update the commit now and close the sul-solr pr's.

**Comment by camillevilla** [comment]
👍 works fine for me now

**Pull request merged**